dockerscripts/docker-entrypoint.sh

```diff
@@ -37,7 +37,76 @@ docker_secrets_env() {
     fi
 }
 
+## Create UID/GID based on available environment variables.
+docker_set_uid_gid() {
+    if  [ -z "$MINIO_GID" ] ||  [ -z "$MINIO_UID" ] ; then
+	    addgroup -g 190 -S minio && adduser -u 190 -S -G minio minio
+    else
+        addgroup -g "$MINIO_GID" -S minio && adduser -u "$MINIO_UID" -S -G minio minio
+    fi
+}
+
+# switch to non root (minio) user if there isn't an existing .minio.sys directory or
+# if its an empty data directory. We dont switch to minio user if there is an existing
+# data directory, this is because we'll need to chown all the files -- which may take
+# too long.
+docker_switch_non_root() {
+    for var in "$@"
+    do
+        if [ "${var}" = "minio" ] || [ "${var}" = "server" ] ; then
+            continue
+        fi
+
+        if [ "${var}" = "gateway" ]; then
+            mode="${var}"
+            continue
+        fi
+        
+        # Updating mode to "gateway nas"
+        if [ "${var}" = "nas" ]; then
+            mode="${mode}"" ${var}"
+            continue
+        fi
+
+
+        # If mode is gateway set owner as minio and break       
+        if [ "${mode}" = "gateway" ]; then
+            owner="minio"
+            break
+        fi
+        
+        # strip scheme and hostname if present in an arg
+        if [ "${var:0:4}" == "http" ]; then
+            var="/$(printf -- "%s" "${var}" | cut -d/ -f4-)"
+        fi   
+
+        if [ -d "${var}" ]; then
+            if ! [ -d "${var}/.minio.sys" ]; then
```

Comment:
We need to check the owner of the directory `.minio.sys` if it exists, and then switch to minio user if owner is `$MINIO_UID`